apollo-backend/src/main/java/io/logz/apollo/controllers/DeploymentController.java

```diff
@@ -164,4 +166,22 @@ public void cancelDeployment(int id, Req req) {
             lockService.releaseLock(lockName);
         }
     }
+
+    @LoggedIn
+    @POST("/deployment-history")
+    public void fetchPaginatedDeploymentHistory(Boolean descending, int pageNumber, int pageSize,  String searchTerm, Req req) {
+
+        DeploymentHistory deploymentHistory = new DeploymentHistory();
```

Comment:
Maybe create a constructor with the relevant parameters instead of calling all the seters. That way you won't be able to miss a parameter that is necessary to your use.
After you will do this you will be able to remove all the seters.

Reply:
done(: